Sources/race.swift

```diff
@@ -12,6 +13,16 @@
  - Warning: If any of the provided promises reject, the returned promise is rejected.
 */
 public func race<T>(promises: Promise<T>...) -> Promise<T> {
+    return try! race(promises)  // race only throws when the array param is empty, which is not
+                                // possible from this variadic paramater version
+}
+
+public func race<T>(promises: [Promise<T>]) throws -> Promise<T> {
+    guard !promises.isEmpty else {
+        let message = "Cannot race with an empty list of runners (Promises)"
+        throw NSError(domain: PMKErrorDomain, code: PMKInvalidUsageError, userInfo: ["messaage": message])
+    }
```

Comment:
Not sure how this should be handled, I'm unable to make this immediately resolve in the case of an empty array because we cannot instantiate a `T` for a use like `return Promise<T>(value: T())`. This forces me to `throw` an error or return an optional promise `Promise<T>?`, but this seemed less desirable.